pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -707,6 +717,22 @@ void PdBlock_iterBegin(JSOBJ _obj, JSONTypeContext *tc) {
 
     for (i = 0; i < PyObject_Length(arrays); i++) {
         array = PyList_GET_ITEM(arrays, i);
+        // tz information is lost when dt64tz is converted
+        // to numpy arrays.
+        // TODO(lithomas1): Patch column_arrays(actually values_for_json)
```

Comment:
How hard is it to patch values_for_json instead? We have a few branches in this function already would be ideal to not add more